views/js/runner/proxy/offline/proxy.js

```diff
@@ -0,0 +1,482 @@
+                        result.testContext = {
+                            state: 4
+                        };
+
+                        if (self.isOffline()) {
+                            return new Promise(function(resolve, reject) {
+                                offlineSyncModal(self)
+                                    .on('proceed', function() {
+                                        self.syncData()
+                                            .then(function() {
+                                                return resolve(result);
+                                            })
+                                            .catch(reject);
+                                    })
+                                    .on('secondaryaction', function() {
+                                        self.initiateDownload()
+                                            .catch(reject);
+                                    });
+                            }).catch(function(error) {
+                                feedback(null, {
+                                    timeout: -1,
+                                    popup: true
+                                }).error(error);
+
+                                return { success: false };
+                            });
+                        } else {
+                            return self.syncData().then(function() {
+                                return result;
+                            }).catch(function(error) {
+                                feedback(null, {
+                                    timeout: -1,
+                                    popup: true
+                                }).error(error);
+
+                                return { success: false };
+                            });
+                        }
+                    }
+
+                    // try the navigation if the actionParams context meaningful data
+                    if (actionParams.direction && actionParams.scope) {
+                        self.offlineNavigator
+                            .setTestData(testData)
+                            .setTestContext(testContext)
+                            .setTestMap(testMap)
+                            .navigate(
+                                actionParams.direction,
+                                actionParams.scope,
+                                actionParams.ref,
+                                actionParams
+                            ).then(function(res) {
+                                newTestContext = res;
+
+                                if (
+                                    !newTestContext
+                                    || !newTestContext.itemIdentifier
+                                    || !self.hasItem(newTestContext.itemIdentifier)
+                                ) {
+                                    throw errorProvider.offlineNavError;
```

Comment:
I'm wondering if this is a good practice, as the error context could be corrupted.